plugins/module_utils/network/eos/providers/cli/config/bgp/neighbors.py

```diff
@@ -180,12 +177,20 @@ def _render_default_originate(self, item, config=None):
 
     def _render_graceful_restart(self, item, config=None):
         cmd = "neighbor %s graceful-restart" % item["neighbor"]
-        if item["activate"] is False:
-            if not config or cmd in config:
-                cmd = "no %s" % cmd
+        if config:
+            config_el = [x.strip() for x in config.split('\n')]
+        if item["graceful_restart"] is False:
+            no_cmd = "no " + cmd
+            if config:
+                if no_cmd in config_el:
+                    return 
+                else:
+                    return no_cmd
+            else:
+                return no_cmd 
+        else:
+            if not config or cmd not in config_el:
                 return cmd
```

Comment:
I think this should preserve 1) checking the whole line and not just a subset and 2) returning the command only when it does not exist in the config. I'm not completely sure, though, as the new logic is a little complex

```suggestion
        if item["graceful_restart"] is False:
            cmd = "no " + cmd
        if config:
            config_el = [x.strip() for x in config.split('\n')]
            if cmd in config_el:
                return
        return cmd
```